### PR TITLE
New Patient Grouping and Priority examples

### PR DESCRIPTION
**Pull request by jacob-morrison-22**

### CODE REVIEW

File: canvas_workflow_helpers/protocols/patient_grouping.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            return True if state == 'rejected' else False
+
+        return False
+
+    def compute_results(self) -> ProtocolResult:
+        result = ProtocolResult()
+        result.status = STATUS_NOT_APPLICABLE
+       
+        patient_key = self.patient.patient['key']
+        patient_group = self.patient.groups.filter(name=self.GROUP_NAME).records
```
Comment (aduane) [inline canvas_workflow_helpers/protocols/patient_grouping.py:72]: I don't think we should filter by name. Let's just hardcode the UUID. That way, we don't have to include any group information as context in the protocol.

**Review by aduane — CHANGES_REQUESTED** [review]

File: canvas_workflow_helpers/protocols/patient_grouping.py (hunk 1 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_grouping.py]: hardcoded uuid for this protocol

**Review by aduane — CHANGES_REQUESTED** [review]
I don't think we should put this customer-specific protocol in our open source repo.

**Comment by jacob-morrison-22** [comment]
> I don't think we should put this customer-specific protocol in our open source repo.

when i spoke with @eabruzzese about this he said as long as the protocol is generic enough, it should go in here as an example. I removed the identifying information, does this work for you? @aduane

File: canvas_workflow_helpers/protocols/patient_grouping.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+
+class PatientGrouping(ClinicalQualityMeasure):
+    """
+    Protocol that updates a patients membership in a group depending on a given consent. 
+    This particular example is of an opt-out based group membership.
+    """
+
+    # Consent codes can be found in the Admin view
+    CONSENT_CODE = '<CODING_FOR_CONSENT>'
```
Comment (aduane) [inline canvas_workflow_helpers/protocols/patient_grouping.py:24]:
```suggestion
    CONSENT_CODE = 'A0001' # replace with your opt-out consent's coding
```

File: canvas_workflow_helpers/protocols/patient_grouping.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return False
+
+    def compute_results(self) -> ProtocolResult:
+        result = ProtocolResult()
+        result.status = STATUS_NOT_APPLICABLE
+       
+        patient_key = self.patient.patient['key']
+
+        # Get this UUID from the api_patientgroup.externally_exposable_id field
+        patient_group_uuid = '<UUID_OF_PATIENT_GROUP>'
```
[the remark this answers is not in the document]
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_grouping.py]: done

File: canvas_workflow_helpers/protocols/patient_grouping.py (hunk 2 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_grouping.py]: done

**Review by aduane — APPROVED** [review]

File: canvas_workflow_helpers/protocols/patient_priority.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        result = ProtocolResult()
+        last_risk_score = self.last_risk_score()
+        if last_risk_score:
+            if last_risk_score in self.SATISFIED_VALUES:
+                result.status = STATUS_SATISFIED
+            else:
+                result.status = STATUS_DUE
+
+        return result
+
```
Comment (rmagier1) [inline canvas_workflow_helpers/protocols/patient_priority.py:97]: Add a new line

File: canvas_workflow_helpers/protocols/patient_priority.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+)
+
+from canvas_workflow_kit.value_set.value_set import ValueSet
+
+
+class QuestionnaireRisk(ValueSet):
+
+    VALUE_SET_NAME = '@Risk Questionnaire'
+
+    INTERNAL = {'Risk_Stratification'}
```
Comment (rmagier1) [inline canvas_workflow_helpers/protocols/patient_priority.py:21]: Can you add a comment on the purpose of this valueset. What is the customer supposed to replace these values with?

File: canvas_workflow_helpers/protocols/patient_priority.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    SATISFIED_VALUES = ('GREEN', 'YELLOW')
+
+    class Meta:
+        title = 'Patient Priority'
+
+        version = '1.0.0'
+
+        description = (
+            'Protocol that displays patient priority based on Risk Questionnaire. '
+            'There is currently no action to take, so no recommendations are made.')
```
Comment (rmagier1) [inline canvas_workflow_helpers/protocols/patient_priority.py:40]:
```suggestion
            'Protocol that displays patient priority based on Risk Questionnaire. '
            'If the patient does not have a Risk Questionnaire, there is no action '
            'to take, so no recommendations are made.')
```

File: canvas_workflow_helpers/protocols/patient_priority.py [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+        result = ProtocolResult()
+        last_risk_score = self.last_risk_score()
+        if last_risk_score:
+            if last_risk_score in self.SATISFIED_VALUES:
+                result.status = STATUS_SATISFIED
+            else:
+                result.status = STATUS_DUE
+
+        return result
```
Comment (rmagier1) [inline canvas_workflow_helpers/protocols/patient_priority.py:96]: Should we not return the result if there is no risk score?

Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py:96]: Patterns I've seen in other protocols is to always return result here

File: canvas_workflow_helpers/protocols/patient_priority.py (hunk 6 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py:40]: I can just remove the wording about recommendations, its kinda unnecessary and might confuse someone looking at this protocol as an example

File: canvas_workflow_helpers/protocols/patient_priority.py (hunk 4 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py:96]: this already has a single new line at the end, should there be two?

File: canvas_workflow_helpers/protocols/patient_priority.py (hunk 5 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py:22]: done

File: canvas_workflow_helpers/protocols/patient_priority.py (hunk 6 above)
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py]: removed mentions of recommedations

File: canvas_workflow_helpers/protocols/patient_priority.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    VALUE_SET_NAME = '@Risk Questionnaire'
+
+    INTERNAL = {'Risk_Stratification'}
+
+    
+
+class PatientPriority(ClinicalQualityMeasure):
+    """
+    Protocol that displays patient priority based on latest risk stratification from Internal @Risk questionnaire, 
+    if the patient has any. There is currently no action to take, so no recommendations are made.
```
[the remark this answers is not in the document]
Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py]: removed mentions of recommedations

File: canvas_workflow_helpers/protocols/patient_priority.py (hunk 5 above)
Comment (rmagier1) [reply canvas_workflow_helpers/protocols/patient_priority.py:22]: It needs to be more specific. What columns or fields in the questionnaire loader did those come from

Comment (rmagier1) [reply canvas_workflow_helpers/protocols/patient_priority.py:22]: 1. Comment for the valueset needs to be like 
```
""" 
  This ValueSet will define the specific Questionnaire that is needed to determine patient's priority. 
"""
```
2.   `# VALUE_SET_NAME can be set to a display name for the Questionnaire we are trying to identify`
3.   `# The name of this variable needs to match the coding system of the questionnaire, while the single value in the set represents the questionnaire's code. These values were passed during Questionnaire Loading`

Comment (jacob-morrison-22) [reply canvas_workflow_helpers/protocols/patient_priority.py:28]: done

**Pull request merged**